### PR TITLE
220 allow admin interaction with bug report

### LINKED ISSUE #220: Allow for the admin to interact with the bug report.
- [x] Add a form to edit the bug report (changing status, adding comment, changing severity, changing type) Possibly a modal

Can only be done after messaging system is implemented: (#195)
- [x] Add an optional form with textarea to send a message to the user who reported the bug

- [x] Upon setting a bug report on resolved, automatically send a notification to the user who reported it to say that the bug has been resolved

### PR DESCRIPTION
**Pull request by LaEve**
all requirements are met

Resolves #220

### CODE REVIEW

File: resources/js/components/modals/EditBugReport.vue [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            bugSeverity: BUG_SEVERITY,
+            bugStatus: BUG_STATUS,
+            message: {
+                message: 'Your bug report has been resolved!',
+            }
+        }
+    },
+    methods: {
+        updateBugReport() {
+            let self = this;
```
Comment (MJZwart) [inline resources/js/components/modals/EditBugReport.vue:91]: I think you don't use self, so can delete this line

File: routes/api.php [hunk 2]
```diff
@@ -76,10 +77,17 @@
     Route::post('/search', [UserController::class, 'searchUser']);
     Route::post('/register/confirm', [RegisteredUserController::class, 'confirmRegister']);
 
-    Route::post('/bugreport', [BugReportController::class, 'store']);
-    Route::get('/bugreports', [BugReportController::class, 'getAllBugs']);
+    Route::put('/bugreport', [BugReportController::class, 'update']);
```
Comment (MJZwart) [inline routes/api.php:80]: Can delete this one, it's already in resource

**Review by MJZwart — CHANGES_REQUESTED** [review]
Couple small things, and the question, did you already do some of the i18n translations? I'm missing a bunch, if you don't know what to write, just write whatever in `en.json` and I'll fix it myself later.

**Comment by LaEve** [comment]
everything is changed according to the requests

File: app/Http/Controllers/BugReportController.php [hunk 3]
```diff
@@ -19,7 +20,23 @@ public function store(StoreBugReportRequest $request): JsonResponse{
...
+    public function update(UpdateBugReportRequest $request, $id): JsonResponse
+    {
+        $validated = $request->validated();
+        
+        $bugReport = BugReport::find($id);
+        $bugReport->type = $validated['type'];
+        $bugReport->severity = $validated['severity'];
+        $bugReport->admin_comment = $validated['admin_comment'];
+        $bugReport->status = $validated['status'];
+        $bugReport->save();
```
[the remark this answers is not in the document]
Comment (MJZwart) [reply app/Http/Controllers/BugReportController.php:30]: You can delete `$bugReport->save()` now

**Review by MJZwart — CHANGES_REQUESTED** [review]

**Pull request merged**